packages/app/src/utils/dynamicUI/extractDynamicConfig.ts

```diff
@@ -166,6 +200,103 @@ function extractDynamicConfig(
     },
     [],
   );
+  config.menuItems = Object.entries(frontend).reduce<GenericMenuItem[]>(
```

Comment:
Can we extract this reduce function? We can also do that as a follow up for the other config objects.

```
config.menuItems = extractMenuItems(...) // or something similar
```